### PR TITLE
test: widget integration tests

### PR DESCRIPTION
**Pull request by just-toby**
adds e2e tests for basic inputs on the Swap Widget

the tests have some limitations because they use goerli tokens. most notable, the TDP test can't actually change the inputs.
we also can't switch chains in these

depends on https://github.com/Uniswap/interface/pull/6176

### CODE REVIEW

File: cypress/e2e/swap-widget.cy.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    // select token...
+    cy.contains(outputText).click()
+
+    cy.get('body')
+      .then(($body) => {
+        if ($body.find(getTestSelector('TokenSafetyWrapper')).length) {
+          return 'I understand'
+        }
+
+        return 'You pay' // Just click on a random element as a no-op
```
Comment (vm) [inline cypress/e2e/swap-widget.cy.ts:27]: you can just return undefined here imo and have it check for undefined in the next statement

File: cypress/e2e/swap-widget.cy.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      })
+      .then((selector) => {
+        cy.contains(selector).click()
+      })
+
+    // token selector should close...
+    cy.contains('Search name or paste address').should('not.exist')
+
+    cy.get(getClassContainsSelector('ReverseButton')).first().click()
```
Comment (vm) [inline cypress/e2e/swap-widget.cy.ts:36]: why do you click this? should there be tests based on hitting it?

File: cypress/e2e/swap-widget.cy.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+describe('swap widget integration tests', () => {
+  const verifyInputToken = (inputText: string) => {
+    cy.get(getClassContainsSelector('TokenButtonRow')).first().contains(inputText)
+  }
+
+  const verifyOutputToken = (outputText: string) => {
+    cy.get(getClassContainsSelector('TokenButtonRow')).last().contains(outputText)
+  }
+
+  const selectOutputAndSwitch = (outputText = 'UNI') => {
```
Comment (vm) [inline cypress/e2e/swap-widget.cy.ts:15]: i don't think it's worth defaulting this param

File: cypress/e2e/swap-widget.cy.ts (hunk 2 above)
Comment (just-toby) [reply cypress/e2e/swap-widget.cy.ts:36]: yes, all the tests that call this function expect it to switch the inputs

File: cypress/utils/index.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export const getTestSelector = (selectorId: string) => `[data-testid=${selectorId}]`
 
 export const getTestSelectorStartsWith = (selectorId: string) => `[data-testid^=${selectorId}]`
+
+export const getClassContainsSelector = (selectorId: string) => `[class*=${selectorId}]`
```
[the remark this answers is not in the document]
Comment (just-toby) [reply cypress/utils/index.ts:5]: cypress actually discourages using classes as test selectors, so i don't think it does (maybe i just couldn't find it though)

[the remark this answers is not in the document]
Comment (vm) [reply cypress/utils/index.ts:5]: oh interesting. should we consider not using the classes then?

File: cypress/e2e/swap-widget.cy.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    // token selector should close...
+    cy.contains('Search name or paste address').should('not.exist')
+
+    cy.get(getClassContainsSelector('ReverseButton')).first().click()
+  }
+
+  describe('widget on swap page', () => {
+    it('should have the correct default input/output and token selection should work', () => {
+      cy.visit('/swap', { featureFlags: [FeatureFlag.swapWidget] }).then(() => {
+        cy.wait('@eth_blockNumber')
```
Comment (vm) [inline cypress/e2e/swap-widget.cy.ts:44]: why is this only for the first test? should this be a precondition?

File: cypress/e2e/swap-widget.cy.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+        selectOutputAndSwitch('UNI')
+
+        verifyInputToken('UNI')
+        verifyOutputToken('ETH')
+      })
+    })
+
+    it('should have the correct default input from URL params ', () => {
+      cy.visit(`/swap?inputCurrency=${UNI_GOERLI}`, {
```
Comment (vm) [inline cypress/e2e/swap-widget.cy.ts:56]: can you put this in beforeEach?

File: cypress/e2e/swap-widget.cy.ts (hunk 2 above)
Comment (vm) [reply cypress/e2e/swap-widget.cy.ts:38]: i might suggest adding an 'exists' case to make sure it actually reversed

File: cypress/e2e/swap-widget.cy.ts (hunk 5 above)
Comment (just-toby) [reply cypress/e2e/swap-widget.cy.ts:44]: yeah i think it's fine to use this in every test - it's just a proxy for "page loaded" which seemed to matter for the first test (only in the GH action) --- although maybe this was just an issue with the flakiness yesterday

File: cypress/e2e/swap-widget.cy.ts (hunk 6 above)
Comment (just-toby) [reply cypress/e2e/swap-widget.cy.ts:60]: i didn't use `beforeEach` for this because it's different for each test!

**Review by vm — APPROVED** [review]
i still think there's a way to abstract the feature flag part of this but don't feel strongly

File: cypress/e2e/swap-widget.cy.ts (hunk 2 above)
Comment (just-toby) [reply cypress/e2e/swap-widget.cy.ts:38]: these are checked at the callsites

**Pull request merged**